### PR TITLE
feat(loss_metrics):add loss & metrics

### PR DESCRIPTION
**Pull request by teamwong111**
1. add focal loss、lovasz loss、dice loss、boundary loss
2. cross entropy loss use tf.keras.losses.SparseCategoricalCrossentropy
3. IOU use tf.keras.metrics.MeanIoU

### CODE REVIEW

File: segelectri/loss_metrics/loss.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    def __init__(self, gamma=2.0, alpha=0.25, **kwargs):
+        super(FocalLoss, self).__init__(**kwargs)
+        self.gamma = gamma
+        self.alpha = alpha
+
+    def call(self, y_true, y_pred):
+        y_true = tf.one_hot(y_true, y_pred.shape[3])
+        y_pred += K.epsilon()
+        ce = -y_true * np.log(y_pred)
+        weight = np.power(1 - y_pred, self.gamma) * y_true
```
[the remark this answers is not in the document]
Comment (teamwong111) [reply segelectri/loss_metrics/loss.py]: 已替换

File: test/test_loss_metrics/test_loss.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+class TestLoss(tf.test.TestCase):
+    def test_focal_loss(self):
+        y_true = tf.random.uniform((2, 512, 512),
+                                   minval=0,
+                                   maxval=3,
+                                   dtype=tf.int64)
+        y_pred = tf.random.uniform((2, 512, 512, 3),
+                                   minval=0,
+                                   maxval=1,
+                                   dtype=tf.float32)
```
[the remark this answers is not in the document]
Comment (teamwong111) [reply test/test_loss_metrics/test_loss.py]: 已修改

**Comment by chong1144** [comment]
LGTM
